reasoner/planner/ReasonerPlanner.java

```diff
@@ -128,8 +130,19 @@ private static boolean notGeneratedByResolvable(Resolvable<?> resolvable, Variab
                 .allMatch(constraint -> constraint.isThing() && constraint.asThing().isValue());
     }
 
+    public static boolean canBypassReasoning(Concludable concludable, Set<com.vaticle.typedb.core.traversal.common.Identifier.Variable.Retrievable> boundVariables, boolean isExplainEnabled) {
+        return !isExplainEnabled && !concludable.isHas() &&
+                concludable.generating().isPresent() && boundVariables.contains(concludable.generating().get().id());
+    }
+
     public Set<CallMode> triggeredCalls(Concludable concludable, Set<Variable> mode, @Nullable Set<ResolvableConjunction> dependencyFilter) {
         Set<CallMode> calls = new HashSet<>();
+
+        // Don't trigger reasoning if we can just look it up
```

Comment:
this comment is inaccurate - the planner doesn't 'trigger' reasoning, it models if it 'will' be triggered in the during execution.

Reply:
removed